### ISSUE TITLE
Reset shelf

### ISSUE THREAD

**Issue by knjk04** [Project-Books/book-project#64]
**Is your feature request related to a problem? If so, please describe.**
It may be annoying if a user has a lot of books in a particular shelf and they want to delete all of the books in that shelf.

**Describe the solution you'd like**
- Add a reset button underneath the grid
- Bring up a dialog window after the user presses the 'reset' button
   - The dialog window should include the following text: `Are you sure you want to reset all of the books in [Shelf name]? This is irreversible.`
   - Options: [yes], [no]
   - Apply `delete` (see `BookService`) for every book in that shelf if and only if the user presses the 'yes' button

**Comment by YBrecher** [comment]
Can I be assigned to this issue?

**Comment by knjk04** [comment]
Sure, thanks!

**Comment by YBrecher** [comment]
![Screenshot (2)](https://user-images.githubusercontent.com/47635350/89065838-66886480-d33a-11ea-93ff-f14fd5a87e3f.png)

This is what I have so far, I still need to set it up so that it displays the correct shelf in the message but I am focusing on getting the basic functionality working first

**Comment by knjk04** [comment]
Ah, OK.


Apologies, this issue is trickier. I believe it should be easier once issue #114 is resolved. We haven't yet come up with a good solution to it, so I would advise waiting until that's resolved. I'm sorry about this.

**Comment by knjk04** [comment]
When this is unblocked, our new confirmation dialog component can be used.

**Comment by knjk04** [comment]
#114 has been merged in, so this is now unblocked.

@YBrecher Are you still happy to work on this?

**Comment by YBrecher** [comment]
Unfortunately, I will not have time to work on this anymore. Can I be unassigned from it?

**Comment by knjk04** [comment]
Sure, that's fine.

**Comment by shivansh29** [comment]
Can I be assigned to this issue??
I'd like to work on it

**Comment by knjk04** [comment]
It's both because a frontend and backend change is required. However, one or the other can be done if we split this into two issues. Unfortunately, this issue is currently blocked. Feel free to request to be assigned to an unblocked and unassigned issue
